onnxruntime/core/session/inference_session.cc

```diff
@@ -52,7 +54,37 @@
 using namespace ONNX_NAMESPACE;
 
 namespace onnxruntime {
+namespace {
+template <typename T>
+const T* GetDateFormatString();
+
+template <>
+inline const char* GetDateFormatString<char>() {
+  return "%Y-%m-%d_%H-%M-%S";
+}
+#ifdef _WIN32
+template <>
+inline const wchar_t* GetDateFormatString<wchar_t>() {
+  return L"%Y-%m-%d_%H-%M-%S";
+}
+#endif
+template <typename T>
+inline std::basic_string<T> GetCurrentTimeString() {
+  auto now = std::chrono::system_clock::now();
```

Comment:
Would be nicer (in a separate PR) to use LoggingManager::GetTimestamp and date::operator<< (see ostream_sink.cc for an example) to simplify this and match the log file timestamp format.